### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by fbeltrao**
Initial version of the sample application

### CODE REVIEW

File: src/Sample.Common/Sample.Common.csproj [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="3.1.0" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="3.1.0" />
+    <PackageReference Include="Microsoft.ApplicationInsights.AspNetCore" Version="2.12.0" />
+    <PackageReference Include="Microsoft.ApplicationInsights" Version="2.12.0" />
+    <PackageReference Include="Microsoft.ApplicationInsights.WorkerService" Version="2.12.0" />
+    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="3.1.0" />
+    <PackageReference Include="Microsoft.Extensions.Options.ConfigurationExtensions" Version="3.1.0" />    
+    <PackageReference Include="System.Text.Json" Version="4.7.0" />
+
+    <PackageReference Include="OpenTelemetry" Version="0.2.0-alpha.151" />
```
Comment (SergeyKanzhelev) [inline src/Sample.Common/Sample.Common.csproj:21]: do you need to add it explicitly?

File: src/Sample.TimeApi/appsettings.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        "Sample.TimeApi.Controllers.TimeController": "Information"
+      }
+    }
+  },
+  "AllowedHosts": "*",
+  "SqlConnectionString": "server=localhost;user id=sa;password=Pass@Word1;",
+  "SampleApp": {
+    "UseApplicationInsights": false,
+    "UseOpenTelemetry": true,
+    "ApplicationInsightsInstrumentationKey": "",
```
Comment (SergeyKanzhelev) [inline src/Sample.TimeApi/appsettings.json:21]: why it's not configured in `ApplicationInsights` section?

File: src/Sample.Common/EnqueuedMessage.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+//using Newtonsoft.Json;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Sample.Common
+{
+    public class EnqueuedMessage
+    {
+        [JsonPropertyName("day")]
+        //[JsonProperty("day", NullValueHandling = NullValueHandling.Ignore)]
```
Comment (SergeyKanzhelev) [inline src/Sample.Common/EnqueuedMessage.cs]: do you need these commented lines?

File: src/Sample.Common/TraceParent.cs [hunk 4]
```diff
@@ -0,0 +1,82 @@
+﻿//using System;
```
Comment (SergeyKanzhelev) [inline src/Sample.Common/TraceParent.cs]: commented out file? Do you need it?

**Review by SergeyKanzhelev — APPROVED** [review]
I glanced over. Looks OK in general.

**Comment by SergeyKanzhelev** [comment]
@lmolkova do you want to take a look?

File: src/Sample.Common/EnqueuedMessage.cs (hunk 3 above)
Comment (fbeltrao) [reply src/Sample.Common/EnqueuedMessage.cs]: They have been removed

File: src/Sample.Common/TraceParent.cs (hunk 4 above)
Comment (fbeltrao) [reply src/Sample.Common/TraceParent.cs]: Don't need it anymore. Thanks

File: src/Sample.Common/Sample.Common.csproj (hunk 1 above)
Comment (fbeltrao) [reply src/Sample.Common/Sample.Common.csproj:21]: What do you mean? The OpenTelemetry version?

File: src/Sample.TimeApi/Data/OpenTelemetryCollectingDeviceRepository.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            var span = this.tracer.StartSpan(nameof(GetTimeFromSqlAsync), SpanKind.Client);
+            try
+            {
+                FailGenerator.FailIfNeeded(1);
+
+                return await this.repository.GetTimeFromSqlAsync();
+            }
+            catch (Exception ex)
+            {
+                span.SetAttribute("error", true);
```
Comment (fbeltrao) [inline src/Sample.TimeApi/Data/OpenTelemetryCollectingDeviceRepository.cs:31]: @SergeyKanzhelev 
Was wondering if that is the right way to make the Span as failed. Moreover, for Jaeger I had to add attribute "error"=true in order to show up nicely in the UI.

File: src/Sample.TimeApi/appsettings.json (hunk 2 above)
Comment (fbeltrao) [reply src/Sample.TimeApi/appsettings.json:21]: Do you mean to move the instrumentation key inside the Logging:ApplicationInsights?

File: src/Sample.Common/PromotheusExporterHostedService.cs [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+            foreach (var initializer in initializers)
+            {
+                initializer.Initialize(meterFactory);
+            }
+
+            this.timer = new Timer(CollectMetrics, meterFactory, Timeout.InfiniteTimeSpan, Timeout.InfiniteTimeSpan);
+
+            exporter.Start();
+
+            this.timer.Change(interval, interval);
```
Comment (fbeltrao) [inline src/Sample.Common/PromotheusExporterHostedService.cs:41]: @SergeyKanzhelev 
I had to add this timer here in order to make the Prometheus exporter work. What is the correct way to make it work?

**Review by SergeyKanzhelev — APPROVED** [review]

**Pull request merged**